### PR TITLE
iOS wrapper viewport implementation

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

**Comment by dvoytenko** [comment]
@cramforce @muxin Tests and documentation have been added. PTAL.

File: src/service/viewport-impl.js [hunk 1]
```diff
@@ -1370,6 +1394,215 @@ export class ViewportBindingNaturalIosEmbed_ {
...
+    this.scrollObservable_.add(callback);
+  }
+
+  /** @override */
+  onResize(callback) {
+    this.resizeObservable_.add(callback);
+  }
+
+  /** @override */
+  updateViewerViewport(unusedViewer) {
```
Comment (muxin) [inline src/service/viewport-impl.js]: Is `updateViewerViewport ` still useful in `viewport` class?

File: css/amp.css [hunk 2]
```diff
@@ -47,6 +47,39 @@ body {
...
+  top: 0 !important;
+  left: 0 !important;
+  right: 0 !important;
+  bottom: 0 !important;
+  margin: 0 !important;
+  display: block !important;
+}
+
+i-amp-html-wrapper > body {
+  position: relative !important;
```
Comment (dvoytenko) [inline css/amp.css]: So that `position:absolute` elements are positioned relative to the `body` and not `i-amp-html-wrapper`.

File: src/service/viewport-impl.js (hunk 1 above)
Comment (dvoytenko) [inline src/service/viewport-impl.js]: This is an interface requirement. It obviously not used here. I'm not sure if it's useful for any other type of viewport, but we shouldn't refactor it in this PR. Please take a look and send a separate PR if it's no longer pertinent.

File: test/functional/test-viewport.js [hunk 3]
```diff
@@ -1354,3 +1355,168 @@ describe('ViewportBindingNaturalIosEmbed', () => {
...
+
+  it('should subscribe to scroll and resize events', () => {
+    expect(win.eventListeners.count('resize')).to.equal(1);
+    // Note that scroll event is on the wrapper, and NOT on root or body.
+    expect(win.eventListeners.count('scroll')).to.equal(0);
+    expect(win.document.eventListeners.count('scroll')).to.equal(0);
+    expect(win.document.documentElement.eventListeners.count('scroll'))
+        .to.equal(0);
+    expect(win.document.body.eventListeners.count('scroll'))
+        .to.equal(0);
```
Comment (muxin) [inline test/functional/test-viewport.js]: Is there a way to test `scroll` eventListener on the wrapper?

File: src/service/viewport-impl.js (hunk 1 above)
Comment (muxin) [inline src/service/viewport-impl.js]: OK

File: test/functional/test-viewport.js (hunk 3 above)
Comment (muxin) [inline test/functional/test-viewport.js]: OK~

**Comment by muxin** [comment]
LGTM
